sdk/package/lib/mainFn/CommandController.ts

```diff
@@ -62,16 +62,25 @@ export class CommandController {
               }
               return subc
             })()
-      let childProcess: cp.ChildProcessWithoutNullStreams
+      let childProcess: cp.ChildProcess
       if (options.runAsInit) {
         childProcess = await subc.launch(commands, {
           env: options.env,
         })
       } else {
         childProcess = await subc.spawn(commands, {
           env: options.env,
+          stdio: options.onStdout || options.onStderr ? "pipe" : "inherit",
         })
       }
+
+      const childProcessWithStdio =
+        childProcess as cp.ChildProcessWithoutNullStreams
```

Comment:
❓ Is this safe as a cast to go from ChildProcess to ChildProcessWithoutNullStreams

Reply:
it worked, but realize now it's incorrect and not even needed, removed.